### DISCUSSION TITLE
Configurable static assets path

### DISCUSSION THREAD

**Discussion by jabrks** [remix-run/remix#5063] [Proposals]
When using remix-serve, It's possible to specify a custom path to serve the build directory from with the `publicPath` property in `remix.config.js`.

https://github.com/remix-run/remix/blob/1bdeebf04d72f41dda8952af0aa7f8a786d6aa1a/packages/remix-serve/index.ts#L18-L21

However, no such option exists to do the same for the static files in the public directory.

https://github.com/remix-run/remix/blob/1bdeebf04d72f41dda8952af0aa7f8a786d6aa1a/packages/remix-serve/index.ts#L23

It would be good to have the option to serve these files from a different path if desired.

**Comment by sergiodxa** [comment]
The files in the public directory are served from `/`, if you want to use another path create a folder inside the public folder, for example `public/assets` and they will be served from `/assets`

I’m not sure what more you need unless you want to be able to rename public to something else for another reason.

**Reply by jabrks** [reply]
The use case I had in mind was more along the lines of emulating a production environment in development.

When I deploy my app, I keep the contents of the `public/` directory separate from the rest of the app. Any request to `/assets` is forwarded to the location of the static assets, and everything else is handled by my Remix request handler. I would like to be able to easily replicate this setup locally.

Right now, I've defined a splat route at `/assets` which proxies any request to `/`, but I'd prefer to be able to configure this in the dev server itself, hence this proposal.

**Comment by ryanflorence** [comment]
Use the express adapter and you get full control over your server. https://remix.run/docs/en/v1/other-api/adapter
